embedded/sql/catalog.go

```diff
@@ -383,13 +389,23 @@ func (t *Table) newIndex(unique bool, colIDs []uint32) (index *Index, err error)
 	// having a direct way to get the indexes by colID
 	for _, col := range index.cols {
 		t.indexesByColID[col.id] = append(t.indexesByColID[col.id], index)
+
+		if col.autoIncrement {
+			// cannot technically happen if the table creation, was properly handled. But it's better to handle the error
+			if t.autoIncrementPK {
```

Comment:
probably we can get rid of `autoIncrementPK` at table struct, and if needed it may just be a method, thus ensuring consistency by construction

Reply:
yes, just check if the col or the index exists

Reply:
or maybe even better. We can get rid of both `autoIncrementPK ` and  `autoIncrementCol`. 

Then add a method to check if the the table has an auto incremental column by checking if the `autoIncrementIndex` is set.
And another method to get the autoIncrementCol by getting the index column(s?)